src/filters/ws.rs

```diff
@@ -305,19 +234,24 @@ impl Sink<Message> for WebSocket {
     type Error = crate::Error;
 
     fn poll_ready(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Result<(), Self::Error>> {
-        (*self).with_context(Some(cx), |s| {
-            cvt(s.write_pending(), "websocket poll_ready error")
+        (*self).with_context(Some(cx), |s, cx| {
+            match s.poll_ready(cx.unwrap()) {
+                Poll::Ready(Ok(())) => Poll::Ready(Ok(())),
+                Poll::Ready(Err(e)) => Poll::Ready(Err(crate::Error::new(e))),
+                Poll::Pending => Poll::Pending,
+            }
+            // cvt(s.write_pending(), "websocket poll_ready error")
         })
     }
 
     fn start_send(mut self: Pin<&mut Self>, item: Message) -> Result<(), Self::Error> {
-        match self.with_context(None, |s| s.write_message(item.inner)) {
+        match self.with_context(None, |s, _| s.start_send(item.inner)) {
             Ok(()) => Ok(()),
             // Err(::tungstenite::Error::SendQueueFull(inner)) => {
             //     log::debug!("websocket send queue full");
             //     Err(::tungstenite::Error::SendQueueFull(inner))
             // }
-            Err(::tungstenite::Error::Io(ref err)) if err.kind() == io::ErrorKind::WouldBlock => {
+            Err(tungstenite::Error::Io(ref err)) if err.kind() == io::ErrorKind::WouldBlock => {
```

Comment:
do we need to address io::ErrorKind::WouldBlock ? it seems it's already being checked on [tokio-tungstenite]( https://github.com/snapview/tokio-tungstenite/blob/607b8bf3d801bb14898fe4c5505cc6292950cd7a/src/lib.rs#L291)

Reply:
DIdn't notice, removed